model/response.py

```diff
@@ -75,6 +78,7 @@ def __init__(self, model, message, wild_type_model_id=None):
             else:
                 self.flux = solution.fluxes.to_dict()
                 self.growth = self.flux[constants.MODEL_GROWTH_RATE[model.id]]
+        self.model.objective = self.old_objective
```

Comment:
Is it possible to use the `cobra.Model` context manager here instead of a temporary variable?

    with self.model:
        self.model.objective = ...

Would perhaps be a bit more readable, but I guess it would have to be done for the entire `respond` function.

Reply:
Depends a bit also how the cache is involved. (I didn't dive into the full code now.) If you only load a certain model state from the cache, make changes, compute a solution, and then discard the model; no sweat :)

Reply:
copying the model is an expensive operation and `with self.model:` does that as far as I understood, which is not necessary in this case

Reply:
No, the context manager does not create a copy. It basically records model modifications and reverses them upon leaving the context. So it is much cheaper than a copy.

Reply:
That actually made me curious so I checked and it looks like [they implemented this with a history manager](https://github.com/opencobra/cobrapy/blob/master/cobra/core/model.py#L966), neat. That shouldn't be so expensive.